### ISSUE TITLE
(Feature Request) Add an 'Add to Dictionary' button

### ISSUE THREAD

**Issue by Zishod** [futo-org/android-keyboard#311]
It would be nice to be able to add words to your personal dictionary from the given suggestions. I'm thinking similar to the Blacklist "word" from suggestions pop up, but to add to dictionary instead of Blacklist.

**Comment by McIrco95** [comment]
Being able to add shortcuts in the popup would be great

**Comment by olafur-palsson** [comment]
I'd like a setting where I can choose to add to personal dictionary by tapping the word in the suggestions. I don't have an Icelandic dictionary and I wouldn't mind putting it in slowly over time.

**Comment by rhubear25** [comment]
> I'd like a setting where I can choose to add to personal dictionary by tapping the word in the suggestions. I don't have an Icelandic dictionary and I wouldn't mind putting it in slowly over time.

... Or click on suggestion word (to implement it), then long click/highlight same word in text, & have context menu option to add word to dictionary.

I keep having experience where I'm typing normal word out & suggestions don't "complete/predict" word for me.... Until I've typed complete word.  Ie predictions are just not working.

**Comment by creeper82** [comment]
Is there any chance for this to be developed? For some reason the system dictionary settings screen doesn't have an Add button. Seems like Huawei implemented a custom screen and it lacks this function. Hence I have no option at all to add a custom word, and the keyboard doesn't seem to automatically learn my custom words which I additionally press on the suggestions menu thousands of times

**Comment by creeper82** [comment]
Yes. GBoard added the word somehow when you clicked on it in suggestions. I can't wait for any implementation of this feature to appear in FUTO. I think anyone using a less common language than English would benefit from that

**Comment by intgr** [comment]
Related:
* https://github.com/futo-org/android-keyboard/issues/1009

**Comment by abb128** [comment]
The reason this button isn't added is because it shouldn't be necessary. Automatic word learning is already in the keyboard and the manual button wouldn't be particularly useful over it. See this FAQ entry here: https://docs.keyboard.futo.org/troubleshooting/faq#its-too-inconvenient-to-add-words-to-personal-dictionary

**Comment by Ash-jib-home** [comment]
Adding a medical/technical use case here.
The existing proposal focuses on words already shown in the suggestion strip. It would also be very useful to add the currently typed, selected, or last committed word when it is not among those suggestions.

For specialist terms such as myelosuppression, I may need to save the word deliberately before it is recognised reliably—especially before using voice input. Automatic learning is helpful, but a quick optional Action would provide an immediate manual fallback without leaving the typing flow. Typing such specialist words repeatedly just to trigger automatic learning is also not practical.

**Comment by yassine-safraoui** [comment]
I came here to say the same thing as @Ash-jib-home, although my use case is different : I'm Moroccan, and we usually type arabic Moroccan dialect (that is Darija) in latin script for convenience, it also allows mixing English, french with Darija. So if I can type some word in Darija and have a way to tell FUTO that this is a real word I want you to remember right after I typed it, without having to go into my custom dictionary and add it there, that would be perfect. And by the way, GBoard already has this and it works really well with this workflow, I've been using GBoard for 4-5 years now and this have been one of its strengths for me. The way it works in GBoard is that when you get suggestions, the word you typed appears among suggestions ; If you tap it, it's added as a custom word and the next time you type it or even swipe it and it'll be recognized. And you can also drag and drop the current word suggestion to delete it and free space for one more suggestion. Its really powerful.

**Comment by s-h-a-d-o-w** [comment]
"Automatic word learning is already in the keyboard and the manual button wouldn't be particularly useful over it. See this FAQ entry here"

I missed that feature so that's great to know.

But... comparing it to Gboard (Which I'd love to be able to fully switch from at some point. Swiping is awesome already!), it unfortunately just doesn't feel as convenient. It doesn't seem to recognize the word after the first time it's manually typed and if you misspell something and want to remove it, there's only blacklisting - which seems harsh. It's nice to have long, complex words stored immediately and to be able to change your mind without then possibly having to dig into the blacklist options.

Also, since you require at least one known word to be typed, I guess that's why email addresses don't seem to be saved?
